data-plane/src/features/implicit.rs

```diff
@@ -199,10 +199,10 @@ impl StringExpr {
             } => todo!(),
             StringExpr::First(list) => list
                 .eval(request)
-                .and_then(|v| v.first().cloned().ok_or_else(|| "List is empty.".into())),
+                .and_then(|v| v.first().cloned().ok_or("List is empty.".into())),
```

Comment:
I added these in. It's more idiomatic and a clippy warning. Without the closure, the expression is executed regardless of whether it is needed. (https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call)

Reply:
Fair enough. Don't love how that reads, but take the point 😄